workers/executor/tasks.py

```
@@ -97,6 +99,25 @@ def execute_extraction(self, execution_context_dict: dict) -> dict:
+    # Batch write usage records collected during execution
+    usage_records = result.metadata.get("usage_records", [])
+    if usage_records:
+        try:
+            config = WorkerConfig()
+            with UsageAPIClient(config) as usage_client:
+                usage_client.set_organization_context(context.organization_id)
+                usage_client.bulk_create_usage(
+                    usage_records,
+                    organization_id=context.organization_id,
+                )
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)